### PR TITLE
Create tables

### PR DESCRIPTION
**Pull request by mollyrossow**
@shen-h, This pull request creates the tables associated with data requests and receipts. (The upper left quadrant of our ERD.) Tables included here are:

- data_request
- contact_history
- dataset_request
- dataset
- data_source
- data_receipt
- data_table

There's a lot here but the real logic is in commit 12c3b31. I know you have a good eye for detail for so please check that's I've included all the fields and keys we need here.

### CODE REVIEW

**Comment by mollyrossow** [comment]
@shen-h  I've added the rest of the tables.

File: alembic/versions/0fbe9f4e9934_create_tables.py [hunk 1]
```diff
@@ -0,0 +1,816 @@
...
+    op.create_foreign_key(
+        'etl_output_data_table_fk',
+        'etl_output',
+        'data_table',
+        ['data_table_id'],
+        ['data_table_id'],
+        source_schema=SCHEMA_NAME,
+        referent_schema=SCHEMA_NAME,
+    )
+
```
Comment (shen-h) [inline alembic/versions/0fbe9f4e9934_create_tables.py]: A reminder to also create_foreign_key for the `etl_output` table on the `workflow_id` and `etl_step` fields pointing at the `etl_step_run` table.

File: alembic/versions/0fbe9f4e9934_create_tables.py [hunk 2]
```diff
@@ -0,0 +1,816 @@
...
+        ['data_table_id', 'column_name'],
+        source_schema=SCHEMA_NAME,
+        referent_schema=SCHEMA_NAME,
+    )
+
+    # Create keys on code_frequency.
+    op.create_primary_key(
+        'code_frequency_fk',
+        'code_frequency',
+        ['data_table_id', 'column_name'],
```
Comment (shen-h) [inline alembic/versions/0fbe9f4e9934_create_tables.py]: Should `code` be appended to the compound primary key list?

File: alembic/versions/0fbe9f4e9934_create_tables.py [hunk 3]
```diff
@@ -0,0 +1,816 @@
...
+        source_schema=SCHEMA_NAME,
+        referent_schema=SCHEMA_NAME,
+    )
+
+    # Create keys on codebook.
+    op.create_primary_key(
+        'codebook_pk',
+        'codebook',
+        ['data_table_id', 'column_name'],
```
Comment (shen-h) [inline alembic/versions/0fbe9f4e9934_create_tables.py]: Should `code` be appended to the list as well?

File: alembic/versions/0fbe9f4e9934_create_tables.py [hunk 4]
```diff
@@ -0,0 +1,816 @@
...
+    op.create_table(
+        'dataset',
+        sa.Column('dataset_id', sa.Integer, primary_key=True),
+        sa.Column('title', sa.Text),
+        sa.Column('description', sa.Text),
+        sa.Column('document_link', sa.Text),
+        sa.Column('keywords', sa.Text),
+        sa.Column('category', sa.Text),
+        sa.Column('data_source_id', sa.Integer),
+        sa.Column('data_set_contact', sa.Text),
```
[the remark this answers is not in the document]
Comment (mollyrossow) [reply alembic/versions/0fbe9f4e9934_create_tables.py]: I've changed `dataset` to `data_set`.

File: alembic/versions/0fbe9f4e9934_create_tables.py [hunk 5]
```diff
@@ -0,0 +1,816 @@
...
+    )
+
+    op.create_table(
+        'numeric_column',
+        sa.Column('data_table_id', sa.Integer),
+        sa.Column('column_name', sa.Text),
+        sa.Column('minimum', sa.Integer),
+        sa.Column('maximum', sa.Integer),
+        sa.Column('mean', sa.Integer),
+        sa.Column('median', sa.Integer),
```
[the remark this answers is not in the document]
Comment (mollyrossow) [reply alembic/versions/0fbe9f4e9934_create_tables.py]: Yes. I've changed Integer to Numeric in numeric_column.

File: alembic/versions/0fbe9f4e9934_create_tables.py [hunk 6]
```diff
@@ -0,0 +1,816 @@
...
+    op.create_foreign_key(
+        'etl_input_data_table_fk',
+        'etl_input',
+        'data_table',
+        ['data_table_id'],
+        ['data_table_id'],
+        source_schema=SCHEMA_NAME,
+        referent_schema=SCHEMA_NAME,
+    )
+
```
[the remark this answers is not in the document]
Comment (mollyrossow) [reply alembic/versions/0fbe9f4e9934_create_tables.py]: We have this on the ERD but I think it's not actually a foreign key in our current design. `etl_step_run` isn't unique on `workflow_id` and `etl_step`. It also needs `etl_step_run_date' as part of the composite key. I'm going to leave it like this for now.

File: alembic/versions/0fbe9f4e9934_create_tables.py (hunk 1 above)
Comment (mollyrossow) [reply alembic/versions/0fbe9f4e9934_create_tables.py]: Same issue as with `etl_input`.

File: alembic/versions/0fbe9f4e9934_create_tables.py (hunk 2 above)
Comment (mollyrossow) [reply alembic/versions/0fbe9f4e9934_create_tables.py]: Fixed.

File: alembic/versions/0fbe9f4e9934_create_tables.py (hunk 3 above)
Comment (mollyrossow) [reply alembic/versions/0fbe9f4e9934_create_tables.py]: Fixed.

**Comment by shen-h** [comment]
@mollyrossow Thanks for the updates! I've squash-merged the pull request. I guess we want to keep the branch for future development but please let me know if I should delete it instead. Thanks.

**Pull request merged**